### DISCUSSION TITLE
Valhalla returning options are invalid error

### DISCUSSION THREAD

**Discussion by sayaligo** [valhalla/valhalla#4645] [Q&A] [answered]
I am trying to read the coordinates from kafka and do map matching on them. Kafka sends a single coordinate(lat,long) at a time. My map matching is unable to work on a single coordinate at a time using meili. If i pass bunch of coordinates directly in my program then it works. Is it a limitation that it cant treat single coordinate at a time and needs at least 2 coordinates ?

**Comment by nilsnolde** [comment, marked as answer]
Yes, for map-matching you need minimum 2 coordinates (maybe even 3 or more, but likely only 2). Sounds like you might want to try /locate..

**Reply by nilsnolde** [reply]
I also wouldn't call it a limitation. It's the same as you can't **route** with a single location..

**Reply by sayaligo** [reply]
Thanks for your reply. In that case , how can i use my kafka as a input to my map matching as it will always stream one record at a time ?

**Reply by nilsnolde** [reply]
I have no idea what kafka is.. better suited for smth like stackoverflow IMO.
